src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -461,18 +504,24 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
         let (metadata_bytecode, module_bytecode) = hex_codes::get_publish_code(market_address);
         code::publish_package_txn(&market_signer, metadata_bytecode, vector[module_bytecode]);
 
-        // Charge registrant.
+        // Only charge the registrant if they have to pay for initializing the chat emojis.
+        // Otherwise, waive the register market fee, since the gas costs of initializing them
+        // is roughly the same.
         let registrant_address = signer::address_of(registrant);
-        let can_pay_fee =
-            coin::is_account_registered<AptosCoin>(registrant_address) &&
-            coin::balance<AptosCoin>(registrant_address) >= MARKET_REGISTRATION_FEE;
-        assert!(can_pay_fee, E_UNABLE_TO_PAY_MARKET_REGISTRATION_FEE);
-        aptos_account::transfer(registrant, integrator, MARKET_REGISTRATION_FEE);
+        let fee = if (ensure_chat_emojis_initialized(registry_ref_mut)) {
+            0
+        } else {
+            let can_pay_fee =
+                coin::is_account_registered<AptosCoin>(registrant_address) &&
+                coin::balance<AptosCoin>(registrant_address) >= MARKET_REGISTRATION_FEE;
+            assert!(can_pay_fee, E_UNABLE_TO_PAY_MARKET_REGISTRATION_FEE);
+            aptos_account::transfer(registrant, integrator, MARKET_REGISTRATION_FEE);
+            (MARKET_REGISTRATION_FEE as u128)
```

Comment:
Since now the fee is not necessarily static, how about an `integrator_fee_paid` field in the `MarketRegistrationEvent`

Can be 0 for the first registration, else `MARKET_REGISTRATION_FEE` for all others

Just useful for accounting

Reply:
Marking resolved, slated for unit testing

Reply:
Addressed at 3077b86 on #37